opentelemetry-contrib/src/datadog/mod.rs

```diff
@@ -149,11 +146,9 @@ impl Default for DatadogPipelineBuilder {
 impl DatadogPipelineBuilder {
     /// Create `ExporterConfig` struct from current `ExporterConfigBuilder`
     pub fn install(mut self) -> Result<(sdk::trace::Tracer, Uninstall), Box<dyn Error>> {
-        let exporter = DatadogExporter::new(
-            self.service_name.clone(),
-            self.agent_endpoint.parse()?,
-            self.version,
-        );
+        let endpoint = self.agent_endpoint + self.version.path();
```

Comment:
The isahc's URL parser cannot modify the `Uri`, its `Uri builder` doesn't really support parse from string. We could introduce some other url parser to help validate and modify the endpoint but that feels like a little bit overhead.

Reply:
I think concatenating here is fine, only happens once, and the URI is still parsed below so it's a valid URI after install